File: clients/Python/model/attack_properties.py
class AttackProperties:
    def __init__(self, attack_range, damage, collect_resource):
        self.attack_range = attack_range
        self.damage = damage
        self.collect_resource = collect_resource
    @staticmethod
    def read_from(stream):
        attack_range = stream.read_int()
        damage = stream.read_int()
        collect_resource = stream.read_bool()
        return AttackProperties(attack_range, damage, collect_resource)
    def write_to(self, stream):
        stream.write_int(self.attack_range)
        stream.write_int(self.damage)
        stream.write_bool(self.collect_resource)
    def __repr__(self):
        return "AttackProperties(" + \
            repr(self.attack_range) + "," + \
            repr(self.damage) + "," + \
            repr(self.collect_resource) + \
            ")"

File: clients/Ruby/model/attack_properties.rb
class AttackProperties
    attr_accessor :attack_range
    attr_accessor :damage
    attr_accessor :collect_resource
    def initialize(attack_range, damage, collect_resource)
        @attack_range = attack_range
        @damage = damage
        @collect_resource = collect_resource
    end
    def self.read_from(stream)
        attack_range = stream.read_int()
        damage = stream.read_int()
        collect_resource = stream.read_bool()
        AttackProperties.new(attack_range, damage, collect_resource)
    end
    def write_to(stream)
        stream.write_int(@attack_range)
        stream.write_int(@damage)
        stream.write_bool(@collect_resource)
    end
end
